### ISSUE TITLE
loss

### ISSUE THREAD

**Issue by yanzichuan** [ControlNet/MARLIN#29]
Hello,

I hope you're doing well. I would appreciate your insights regarding some loss values I observed during training. Specifically, I encountered the following values for v_num=12:

val_loss = -1.61
val_d_loss = -1.72
val_d_loss0 = -1.72
val_g_loss = 0.109
val_rec_loss = 0.116
val_adv_loss = -0.00658
I am curious if these values seem reasonable, especially given the negative values for val_loss and val_d_loss. I would appreciate any guidance on whether this behavior is expected or if it may indicate an issue with my setup or training process.

Thank you in advance for your help!

Best regards,
Zack

**Comment by yanzichuan** [comment]
Thank you for the amazing work your team has done and for sharing your contributions in Facial Expression Recognition (FER). I was wondering if it would be possible for you to provide the training and inference code for LP (Linear Probing) or FT (Fine-Tuning) used in your FER experiments. It would be incredibly helpful for furthering my understanding and experiments.

**Comment by ControlNet** [comment]
Hi Zack,

> I am curious if these values seem reasonable, especially given the negative values for val_loss and val_d_loss. I would appreciate any guidance on whether this behavior is expected or if it may indicate an issue with my setup or training process.

The `d_loss` is the WGAN loss which is fine to be negative.

$L_{adv}^{\(d\)} = discriminator(fake\ samples) - discriminator(real\ samples)$

> Code for FER.

The code for FER is very similar to the code for attributes classification, as it's just adding the MARLIN encoder and a linear layer. You can easily convert the code for attribute classification to FER.

**Comment by yanzichuan** [comment]
Thank you sincerely for your timely response and for addressing my questions. I truly appreciate your support and valuable insights.

**Comment by yanzichuan** [comment]
loss_dict = {
            "loss": d_loss + g_loss,
            **{k: v for k, v in d_result.items() if k != "loss"},
            **{k: v for k, v in g_result.items() if k != "loss"},
        }
        is "loss" :  | d_loss | + | g_loss| ?
